messaginginapp/src/main/java/io/customer/messaginginapp/ModuleMessagingInApp.kt

```diff
@@ -12,82 +14,40 @@ import io.customer.sdk.events.Metric
-    private val gistProvider by lazy { diGraph.gistProvider }
+    private val gistProvider: GistProvider
+        get() = diGraph.gistProvider
 
-    private val inAppMessagingManager: InAppMessagingManager = SDKComponent.inAppMessagingManager
+    private val logger = SDKComponent.logger
 
     fun dismissMessage() {
         gistProvider.dismissMessage()
     }
 
     override fun initialize() {
-        inAppMessagingManager.dispatch(InAppMessagingAction.Initialize)
-        initializeGist()
         setupHooks()
-        configureSdkModule(moduleConfig)
-        setupGistCallbacks()
-    }
-
-    private fun configureSdkModule(moduleConfig: MessagingInAppModuleConfig) {
-        moduleConfig.eventListener?.let { eventListener ->
-            gistProvider.setListener(eventListener)
-        }
-    }
-
-    private fun setupGistCallbacks() {
-        gistProvider.subscribeToEvents(onMessageShown = { deliveryID ->
-            inAppMessagingManager.dispatch(InAppMessagingAction.LogEvent("in-app message shown in callback $deliveryID"))
-            eventBus.publish(
-                Event.TrackInAppMetricEvent(
-                    deliveryID = deliveryID,
-                    event = Metric.Opened
-                )
-            )
-        }, onAction = { deliveryID: String, _: String, action: String, name: String ->
-            inAppMessagingManager.dispatch(InAppMessagingAction.LogEvent("in-app message clicked in callback $deliveryID"))
-            eventBus.publish(
-                Event.TrackInAppMetricEvent(
-                    deliveryID = deliveryID,
-                    event = Metric.Clicked,
-                    params = mapOf("action_name" to name, "action_value" to action)
-                )
-            )
-        }, onError = { errorMessage ->
-            inAppMessagingManager.dispatch(InAppMessagingAction.LogEvent("in-app message error occurred $errorMessage"))
-        })
     }
 
     private fun setupHooks() {
         eventBus.subscribe<Event.ScreenViewedEvent> {
-            inAppMessagingManager.dispatch(InAppMessagingAction.SetCurrentRoute(it.name))
             gistProvider.setCurrentRoute(it.name)
         }
 
         eventBus.subscribe<Event.ProfileIdentifiedEvent> {
-            inAppMessagingManager.dispatch(InAppMessagingAction.SetUser(it.identifier))
-            gistProvider.setUserToken(it.identifier)
+            gistProvider.setUserId(it.identifier)
         }
 
         eventBus.subscribe<Event.ResetEvent> {
-            inAppMessagingManager.dispatch(InAppMessagingAction.ClearUser)
-            gistProvider.clearUserToken()
+            logger.debug("Resetting user token")
+            gistProvider.reset()
         }
     }
 
-    private fun initializeGist() {
-        gistProvider.initProvider(
-            application = diGraph.android().application,
-            siteId = moduleConfig.siteId,
-            region = moduleConfig.region.code
-        )
-    }
-
     companion object {
```

Comment:
This may sound a bit nitpicky 😁 But would appreciate if we could avoid placing `companion object` in middle of the class, as it makes the code harder to read IMO. It would be better to place it either at the bottom (preferred) or at the top of the class so it's easier to locate.